collector/collector.go

```diff
@@ -196,7 +197,7 @@ func (c *Collector) addBuildAndJobMetrics(r *Result) error {
 				continue
 			}
 
-			pipeline := *build.Pipeline.Name
+			pipeline, _ := idna.ToASCII(*build.Pipeline.Name)
```

Comment:
I'm not sure if we should be more worried about errors here. The only error case I can think of is malformed UTF-8, which seems like a minimal risk given that we've already read the string in to Go.

Reply:
If we do get an error it will result in a blank string for pipeline though, which will be a whole lot harder to troubleshoot than an explicit error.